### ISSUE TITLE
PTY.open unusable because of fcntl error

### ISSUE THREAD

**Issue by byteit101** [jruby/jruby#5004]
### Environment
* jruby 9.1.15.0 (2.3.3) 2017-12-07 929fde8 OpenJDK 64-Bit Server VM 25.151-b12 on 1.8.0_151-b12 +jit [linux-x86_64]
* Multiple linuxes (RHELs & Debian)

### MRI
```
2.3.0 :001 > require 'pty'
 => true 
2.3.0 :002 > PTY.open{|m,s|}
 => nil 
```

### Actual Behavior
```
jruby-9.1.15.0 :001 > require 'pty'
 => true 
jruby-9.1.15.0 :002 > PTY.open{|m,s|}
NotImplementedError: F_SETFL only supports O_NONBLOCK
	from org/jruby/RubyIO.java:2441:in `fcntl'
	from /home/byteit101/.rvm/rubies/jruby-9.1.15.0/lib/ruby/stdlib/pty.rb:140:in `hack_close_on_exec'
	from /home/byteit101/.rvm/rubies/jruby-9.1.15.0/lib/ruby/stdlib/pty.rb:57:in `block in open'
	from org/jruby/RubyArray.java:1735:in `each'
	from /home/byteit101/.rvm/rubies/jruby-9.1.15.0/lib/ruby/stdlib/pty.rb:54:in `open'
	from (irb):2:in `<eval>'
	from org/jruby/RubyKernel.java:994:in `eval'
	from org/jruby/RubyKernel.java:1292:in `loop'
	from org/jruby/RubyKernel.java:1114:in `catch'
	from org/jruby/RubyKernel.java:1114:in `catch'
	from /home/byteit101/.rvm/rubies/jruby-9.1.15.0/bin/irb:13:in `<main>'

```

**Comment by headius** [comment]
Fallout from adding CLOEXEC support; now it is defined so code runs that didn't before, but not all paths have been fixed to support it.

**Comment by headius** [comment]
Fixed for 9.1.16 by softening the restrictions on fcntl flags. The logic in there is not correct yet, since it doesn't clear flags and only sets specific ones (NONBLOCK and CLOEXEC) but it works properly again (and is no worse than before).
